crates/humanode-runtime/src/lib.rs

```diff
@@ -1502,6 +1502,194 @@ impl_runtime_apis! {
         }
     }
 
+    impl evm_tracing_api::EvmTracingApi<Block> for Runtime {
+        fn trace_transaction(
+            extrinsics: Vec<<Block as BlockT>::Extrinsic>,
+            traced_transaction: &EthereumTransaction,
+            header: &<Block as BlockT>::Header,
+        ) -> Result<(), sp_runtime::DispatchError> {
+            #[cfg(feature = "evm-tracing")]
+            {
+                Executive::initialize_block(header);
+
+                for ext in extrinsics {
+                    match &ext.0.function {
+                        RuntimeCall::Ethereum(transact { transaction }) => {
+                            let tx_hash = &transaction.hash();
+                            if transaction == traced_transaction {
+                                evm_tracer::EvmTracer::default().trace(|| {
+                                    if let Err(err) = Executive::apply_extrinsic(ext) {
+                                        frame_support::log::debug!(
+                                            target: "tracing",
+                                            "Could not trace eth transaction (hash: {}): {:?}",
+                                            &tx_hash,
+                                            err
+                                        );
+                                    }
+                                });
+                            } else if let Err(err) = Executive::apply_extrinsic(ext) {
+                                frame_support::log::debug!(
+                                    target: "tracing",
+                                    "Failed to apply eth extrinsic (hash: {}): {:?}",
+                                    &tx_hash,
+                                    err
+                                );
+                            }
+                        }
+                        _ => {
+                            if let Err(err) = Executive::apply_extrinsic(ext) {
+                                frame_support::log::debug!(
+                                    target: "tracing",
+                                    "Failed to apply non-eth extrinsic: {:?}",
+                                    err
+                                );
+                            }
+                        }
+                    };
+                }
+
+                Ok(())
+            }
+
+            #[cfg(not(feature = "evm-tracing"))]
+            {
+                let _ = extrinsics;
+                let _ = traced_transaction;
+                let _ = header;
+
+                Err(sp_runtime::DispatchError::Other(
+                    "Missing `evm-tracing` compile time feature flag.",
```

Comment:
```suggestion
                    "Missing `evm-tracer` compile time feature flag.",
```

The flag is actually called differently at the `Cargo.toml`

Reply:
`evm-tracing` is a feature, `evm-tracer` is an object that does some logic related with EVM tracing

Reply:
Added a dedicated `evm-tracing` feature at `evm-tracer`